pkg/util/validation/validation.go

```diff
@@ -16,6 +16,67 @@ import (
 // LabelValueMaxLength is a label's max length
 const LabelValueMaxLength int = 63
 
+const (
+	clusterGroupMinOrder = 1
+	clusterGroupMaxOrder = 100
+)
+
+// ValidatePropagationSpec validates a webhook before creation or update.
+func ValidatePropagationSpec(spec policyv1alpha1.PropagationSpec) field.ErrorList {
+	var allErrs field.ErrorList
+	allErrs = append(allErrs, ValidatePlacement(spec.Placement, field.NewPath("spec").Child("placement"))...)
+	return allErrs
+}
+
+// ValidatePlacement validates a webhook before creation or update.
+func ValidatePlacement(placement policyv1alpha1.Placement, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	if placement.ClusterAffinity != nil && placement.OrderedClusterAffinities != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, placement, "orderedClusterAffinities can not co-exist with clusterAffinity"))
+	}
+
+	allErrs = append(allErrs, ValidateClusterAffinity(placement.ClusterAffinity, fldPath.Child("clusterAffinity"))...)
+	allErrs = append(allErrs, ValidateOrderedClusterAffinities(placement.OrderedClusterAffinities, fldPath.Child("orderedClusterAffinities"))...)
+	allErrs = append(allErrs, ValidateSpreadConstraint(placement.SpreadConstraints, fldPath.Child("spreadConstraints"))...)
+	return allErrs
+}
+
+// ValidateClusterAffinity validates a webhook before creation or update.
+func ValidateClusterAffinity(affinity *policyv1alpha1.ClusterAffinity, fldPath *field.Path) field.ErrorList {
+	if affinity == nil {
+		return nil
+	}
+
+	var allErrs field.ErrorList
+	err := ValidatePolicyFieldSelector(affinity.FieldSelector)
+	if err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("fieldSelector"), affinity.FieldSelector, err.Error()))
+	}
+	return allErrs
+}
+
+// ValidateOrderedClusterAffinities validates a webhook before creation or update.
+func ValidateOrderedClusterAffinities(affinities []policyv1alpha1.OrderedClusterAffinityTerm, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	orders := make(map[int32]bool)
+	for index, term := range affinities {
+		if term.Order < clusterGroupMinOrder || term.Order > clusterGroupMaxOrder {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(index).Child("order"), term.Order, "order need to be in the range 1-100"))
```

Comment:
Why not use `+kubebuilder:validation:Maximum=100` and `+kubebuilder:validation: Minimum=1`  ?

Reply:
I think we can do what you say.